pkg/kratos/service.go

```diff
@@ -123,14 +123,19 @@ func (s *Service) CreateBrowserLoginFlow(
 	ctx, span := s.tracer.Start(ctx, "kratos.Service.CreateBrowserLoginFlow")
 	defer span.End()
 
-	flow, resp, err := s.kratos.FrontendApi().
+	request := s.kratos.FrontendApi().
 		CreateBrowserLoginFlow(ctx).
 		Aal(aal).
 		ReturnTo(returnTo).
-		LoginChallenge(loginChallenge).
 		Refresh(refresh).
-		Cookie(httpHelpers.CookiesToString(cookies)).
-		Execute()
+		Cookie(httpHelpers.
+		CookiesToString(cookies))
+
+	if loginChallenge != "" {
+		request.LoginChallenge(loginChallenge)
+	}
+
+	flow, resp, err := request.Execute()
```

Comment:
We don't have a home page, this means that the user should not log in without a login challenge OR a return to. Let's add this logic to the code, I can make this change if you want

Reply:
Right, we should ensure either of the two are set. I was removing the need to pass the login challenge with this change. Because users might land on the "reset password" page, and should be able to login, which was previously failing. I think the reset password page would then be the return to address, so it should be fine as you suggested.

Please, can you add commits to my branch to enforce the requirement to have either the challenge or the return to present @nsklikas ?